pwndbg/commands/context.py

```diff
@@ -851,48 +856,57 @@ def get_thread_status(thread):
 
 
 def context_threads(with_banner=True, target=sys.stdout, width=None):
-    threads = gdb.selected_inferior().threads()[::-1]
+    all_threads = gdb.selected_inferior().threads()[::-1]
+    displayed_threads = all_threads[: int(config_max_threads_display)]
+    num_threads_not_shown = len(all_threads) - len(displayed_threads)
 
-    if len(threads) < 2:
+    if len(displayed_threads) < 2:
         return []
 
-    selected_thread = gdb.selected_thread()
-    selected_frame = gdb.selected_frame()
+    original_thread = gdb.selected_thread()
 
-    out = []
+    out = [pwndbg.ui.banner(f"threads ({len(all_threads)} total)", target=target, width=width)]
     max_name_length = 0
 
-    for thread in threads:
+    for thread in displayed_threads:
         name = thread.name or ""
         if len(name) > max_name_length:
             max_name_length = len(name)
 
-    for thread in threads:
-        thread.switch()
-        frame = gdb.selected_frame()
-
-        selected = " ►" if thread is selected_thread else "  "
-
-        symbol = pwndbg.gdblib.symbol.get(frame.pc())
-        status = get_thread_status(thread)
-
+    for thread in filter(lambda t: t.is_valid(), displayed_threads):
+        selected = " ►" if thread is original_thread else "  "
         name = thread.name if thread.name is not None else ""
         padding = max_name_length - len(name)
+        status = get_thread_status(thread)
 
         line = (
             f" {selected} {thread.global_num}\t"
             f'"{pwndbg.color.cyan(name)}" '
             f'{" " * padding}'
-            f"{status}: {M.get(frame.pc())}"
+            f"{status}: "
         )
-        if symbol:
-            line += f" <{pwndbg.color.bold(pwndbg.color.green(symbol))}> "
+
+        if thread.is_stopped():
+            thread.switch()
+            frame = gdb.selected_frame()
+
+            pc = M.get(frame.pc())
+            symbol = pwndbg.gdblib.symbol.get(frame.pc())
```

Comment:
Maybe:
```py
pc = gdb.selected_frame().pc()
pc, symbol = M.get(pc), pwndbg.gdblib.symbol.get(pc)
```

Reply:
Why 🤔

Reply:
Ah I assume you want to remove the redundant call to `frame.pc()` 👍